### DISCUSSION TITLE
Thicken

### DISCUSSION THREAD

**Discussion by jgomez720** [KittyCAD/modeling-app#499] [New Feature Requests]
### Description
A tool that quickly allows the user to add more thickness to the face or faces of a body. In most cases, you can accomplish the same thing with an extrude, but Thicken is a lot quicker.

### Functionality
1. The user selects the faces they want to thicken.
2. The user selects the "Thicken" button.
3. The user inputs the dimension to thicken by.

OR

1. The user selects the "Thicken" button.
2. The user selects the faces they want to thicken.
3. The user inputs the dimension they want to thicken by.

![Screenshot 2023-09-13 at 2 48 27 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/67a0ccc5-467a-46ee-831c-a4420e9f96f4)
![Screenshot 2023-09-13 at 2 48 42 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/55b77e7c-c836-4e46-9393-84fe388c984a)
![Screenshot 2023-09-13 at 2 48 48 PM](https://github.com/KittyCAD/modeling-app/assets/114548659/2a152e3f-05fd-401d-a464-43e1835658d6)

**Comment by nicboone8** [comment]
I actually associate "thicken" as something you do to lines, not faces. Like if you had a 3D space frame made up of lines and nodes and wanted to convert all the lines into cylinders without having to sweep the same circle profile along all of them. Ideally that tool would also merge all the intersections where the tubes meet.

Having a button to quickly convert a face into a closed sketch that you can manipulate is super valuable. IMO that's what we should have a button for, rather than a thicken extrude. That way you could take that face and also quickly revolve, loft, offset, etc.

I'm accustomed to doing things like this in the 2d sketcher, if you sketch on face then extract the perimeter of that face as lines. Being able to do it quickly from the 3D is compelling though

**Reply by maruel** [reply]
What you are describing in "Extrude / Thin" in other software.

**Reply by jgomez720** [reply]
I also know that as a thin extrude. 

I also agree that converting a face into a closed sketch would be incredible. Something like "covert to sketch entities" or similar you'll see in other programs

**Reply by nicboone8** [reply]
Ok yeah, I articulated that poorly. Extrude thin drags a sketch a specified distance then adds a thickness. Like turning a circle into a walled tube. I meant something more like a multipipe operation, where you're literally inflating the lines
